lighting/material.glsl

```diff
@@ -35,6 +34,8 @@ struct Material {
 
     vec3    position;       // world position of the surface
     vec3    normal;         // world normal of the surface
+    float   sdf;
```

Comment:
Maybe adding `#if defined(RENDER_RAYMARCHING)` and to `raymarching.*` add this tag like with `pbrClearCoat.*` ?

Reply:
It's a good idea, but it makes the include system a bit more brittle. If `material.glsl` is included before `raymarch.glsl`(on the client side), then `RENDER_RAYMARCHING`will not get a chance to be declared, and the taymarching system will break. Let me give it a think and get back to you.

Reply:
@patriciogonzalezvivo due to circular dependencies, this is turning out to be more complicated than hoped.

Take this simple example:
```
#include "lygia/lighting/pbr.glsl"
#include "lygia/lighting/raymarch.glsl"
```
`pbr.glsl` includes `material.glsl` which means `Material` will get initialised before `raymarch.glsl` gets a chance to define `RENDER_RAYMARCHING`.

The reverse option also fails
```
#include "lygia/lighting/raymarch.glsl"
#include "lygia/lighting/pbr.glsl"
```
since `raymarch.glsl` relies on the `pbr` shading function.

if we  really want to use the `#if defined(RENDER_RAYMARCHING)` pattern, the only option I can think of is to require the client to manually define `RENDER_RAYMARCHING` in their code when needed:
```
#define RENDER_RAYMARCHING
#include "lygia/lighting/raymarch.glsl"
#include "lygia/lighting/pbr.glsl"
```

I'm OK with it. Let me know what you think.

Reply:
Good points. So it seams is about managing memory foot print vs easiness. Let's favor easiness and leave room for memory optimization.

What about adding this at the top:

```glsl
#if !defined(MATERIAL_OPT_IN)
#define RENDER_RAYMARCHING
#define SHADING_MODEL_CLOTH
#define SHADING_MODEL_CLEAR_COAT
#define SHADING_MODEL_CLOTH
#define SHADING_MODEL_SPECULAR_GLOSSINESS
...
#endif
``` 
Where by default is permissive and instantiate all the variables on the struct. And then for pro-users that want to be more efficient they need to use `#define MATERIAL_OPT_IN` + the defines they want

Reply:
@patriciogonzalezvivo Sounds like a good compromise! Just pushed to the branch.
BTW it's only needed for `RENDER_RAYMARCHING`. `SHADING_MODEL_*` all live in material sub-system and cause no circular dependency issues.